src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/AnalyzerConfigItemHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+
+using Microsoft.VisualStudio.ProjectSystem.Logging;
+
+namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices.Handlers
+{
+    /// <summary>
+    ///     Handles changes to the &lt;EditorConfigFiles/&gt; items during design-time builds.
+    /// </summary>
+    [Export(typeof(IWorkspaceContextHandler))]
+    internal class AnalyzerConfigItemHandler : AbstractWorkspaceContextHandler, ICommandLineHandler
+    {
+        private readonly UnconfiguredProject _project;
+        private readonly HashSet<string> _paths = new HashSet<string>(StringComparers.Paths);
+
+        [ImportingConstructor]
+        public AnalyzerConfigItemHandler(UnconfiguredProject project)
+        {
+            Requires.NotNull(project, nameof(project));
```

Comment:
When can this happen?

Reply:
This is heavily based on the `AdditionalFilesItemHandler`, right down to the doc comment. It may also handle the evaluation side of things; I'm not entirely sure. :-)

Also, you could conceivably have a target that generates MSBuild items for .editorconfig files, so we can't just consider the evaluation model.

Reply:
Short term this is okay. When I move AdditionalFiles to also respect evaluation, I'll move this one too. 

The reason you want to respect evaluation is for user experience; you drop an editorconfig in the project or a directory - you probably want Roslyn to see the file immediately, not later on when the design-time build has finished.